service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceRegistrar.java

```diff
@@ -80,21 +81,31 @@ private void registerKubernetesService(String application, Metadata<KubernetesMe
                     application + "-" + ipAsSuffix(ipAddress), namespace, null, UUID.randomUUID().toString());
             EndpointAddress endpointAddress = new EndpointAddressBuilder().withIp(ipAddress).withTargetRef(targetRef)
                     .build();
-            Endpoints endpoint = new EndpointsBuilder()
-                    .withNewMetadata().withName(application).withLabels(serviceLabels).endMetadata()
-                    .addToSubsets(new EndpointSubsetBuilder().withAddresses(endpointAddress)
-                            .addToPorts(new EndpointPortBuilder().withPort(8080).build())
-                            .build())
+            EndpointSubset endpointSubset = new EndpointSubsetBuilder().withAddresses(endpointAddress)
+                    .addToPorts(new EndpointPortBuilder().withPort(8080).build())
```

Comment:
I'm wondering about the hardcoded 8080 port. Shouldn't we look at the service object to extract the port?

Reply:
Yes, the hardcoding is not good. We need to pass the port instead.